### PR TITLE
Release-workflow: Bug fix

### PR DESCRIPTION
**Pull request by andersy005**
This fixes https://github.com/pydata/xarray/issues/5232#issuecomment-833729171

### CODE REVIEW

**Comment by andersy005** [comment]
@TomNicholas & @alexamici... If you still want to use this, you may want to delete the old release/tag and re-publish it :( Sorry for the extra work

**Comment by andersy005** [comment]
Also, feel free to merge this at your earliest convenience

File: .github/workflows/pypi-release.yaml [hunk 1]
```diff
@@ -81,7 +81,7 @@ jobs:
 
   upload-to-pypi:
     needs: test-built-dist
-    if: github.event_name == 'release' && startsWith(github.event.ref, 'refs/tags/v')
+    if: github.event_name == 'release' && startsWith(github.ref, 'refs/tags/v')
```
Comment (keewis) [inline .github/workflows/pypi-release.yaml:84]: do we even need the extra condition? `event_name == 'release'` should make that redundant.

Comment (andersy005) [reply .github/workflows/pypi-release.yaml:84]: As long as we are okay with publishing every tag to PyPI, then I think the `event_name` is redundant...

Comment (keewis) [reply .github/workflows/pypi-release.yaml:84]: no, I meant to remove the `startsWith(...)` part: `github.event_name == 'release'` should guarantee that someone created a release through the web page

Comment (andersy005) [reply .github/workflows/pypi-release.yaml:84]: I see... Yes, that's right. We don't need that part.

Comment (andersy005) [reply .github/workflows/pypi-release.yaml]: Thanks for catching that!

**Comment by keewis** [comment]
this should be ready to go (no need to wait on RTD)

**Pull request merged**